### PR TITLE
Custom ssl certs for git

### PR DESCRIPTION
**Pull request by blaster128823782372**

### CODE REVIEW

**Comment by 6543** [comment]
just an idea: could we extend this feature to either load from file or download from url?

File: main.go [hunk 1]
```diff
@@ -83,6 +83,11 @@ func main() {
 			Usage:   "skip tls verification",
 			EnvVars: []string{"PLUGIN_SKIP_VERIFY"},
 		},
+		&cli.StringFlag{
+			Name:    "custom-cert",
+			Usage:   "path to custom cert",
```
Comment (6543) [inline main.go:88]:
```suggestion
			Usage:   "path or url to custom cert",
```

File: main.go [hunk 2]
```diff
@@ -83,6 +83,11 @@ func main() {
...
+			EnvVars: []string{"PLUGIN_CUSTOM_SSL_PATH"},
```
Comment (6543) [inline main.go:89]:
```suggestion
			EnvVars: []string{"PLUGIN_CUSTOM_SSL_PATH", "PLUGIN_CUSTOM_SSL_URL"},
```

File: plugin.go [hunk 3]
```diff
@@ -80,6 +87,53 @@ func (p Plugin) Exec() error {
 
 	return nil
 }
+func customCertHandler(certPath string) *exec.Cmd {
```
Comment (6543) [inline plugin.go:90]: create a new package "cert" to move related functions into?

**Comment by 6543** [comment]
@blaster128823782372 if you need help just ask :)

File: plugin.go (hunk 3 above)
Comment (blaster128823782372) [reply plugin.go:90]: i think moving to a new file is a bit too much. we are not using this functions from anywhere else. So it doesn't make much sense to me.

File: plugin.go [hunk 4]
```diff
@@ -81,6 +88,56 @@ func (p Plugin) Exec() error {
 	return nil
 }
 
+func customCertHandler(certPath string) *exec.Cmd {
+	if IsUrl(certPath) {
+		if downloadCert(certPath) {
+			return setCustomCert("/tmp/customCert.pem")
```
Comment (6543) [inline plugin.go:94]: just add a const for the string `/tmp/customCert.pem` and use it :)

else it's good to go

**Review by 6543 — APPROVED** [review]

**Pull request merged**